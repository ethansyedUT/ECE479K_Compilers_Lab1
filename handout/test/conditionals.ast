#8
_program
  #8
  _class
    Main
    Object
    "conditionals.cl"
    (
    #7
    _method
      main
      Int
      #7
      _block
        #6
        _cond
          #4
          _lt
            #3
            _int
              5
            : Int
            #3
            _int
              7
            : Int
          : Bool
          #5
          _plus
            #4
            _int
              1
            : Int
            #4
            _int
              2
            : Int
          : Int
          #6
          _plus
            #5
            _int
              5
            : Int
            #5
            _int
              7
            : Int
          : Int
        : Int
      : Int
    )
