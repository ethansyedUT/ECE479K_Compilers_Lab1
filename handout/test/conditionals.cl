class Main {
    main() : Int {{
        if true
        then 1 + 2
        else 5 + 7
        fi;

        if false
        then 1 + 2
        else 5 + 7
        fi;
    }};
};