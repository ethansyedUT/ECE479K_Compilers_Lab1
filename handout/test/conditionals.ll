declare i32 @strcmp(i8*, i8*)
declare i32 @printf(i8*, ...)
declare void @abort()
declare i8* @malloc(i32)
define i32 @Main_main() {

entry:
	%vtpm.0 = alloca i32
	%vtpm.1 = icmp slt i32 5, 7
	br i1 %vtpm.1, label %btrue0, label %bfalse0

btrue0:
	%vtpm.2 = add i32 1, 2
	store i32 %vtpm.2, i32* %vtpm.0
	br label %end0

bfalse0:
	%vtpm.3 = add i32 5, 7
	store i32 %vtpm.3, i32* %vtpm.0
	br label %end0

end0:
	%vtpm.4 = load i32, i32* %vtpm.0
	ret i32 %vtpm.4
}

@.str = internal constant [25 x i8] c"Main_main() returned %d\0A\00"
define i32 @main() {

entry:
	%vtpm.5 = call i32 @Main_main(  )
	%vtpm.6 = getelementptr [25 x i8], [25 x i8]* @.str, i32 0, i32 0
	%vtpm.7 = call i32(i8*, ... ) @printf( i8* %vtpm.6, i32 %vtpm.5 )
	ret i32 0
}

