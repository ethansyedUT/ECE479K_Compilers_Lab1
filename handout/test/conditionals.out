Main_main() returned 3
